smart-contracts/assembly/contracts/NFT/NFT-internals.ts

```diff
@@ -1,282 +1,236 @@
 import {
   stringToBytes,
-  bytesToU64,
-  u64ToBytes,
+  bytesToU256,
+  bytesToString,
+  boolToByte,
+  byteToBool,
+  u256ToBytes,
   Args,
-  SafeMath,
 } from '@massalabs/as-types';
-import { Storage, Context, validateAddress } from '@massalabs/massa-as-sdk';
+import { Storage, Context, Address } from '@massalabs/massa-as-sdk';
 
-import {
-  approvedForAllTokenKey,
-  approvedTokenKey,
-  baseURIKey,
-  counterKey,
-  initCounter,
-  nameKey,
-  nft1_ownerOf,
-  ownerKey,
-  ownerTokenKey,
-  symbolKey,
-  totalSupplyKey,
-} from './NFT';
-
-/**
- * Initialize all the properties of the NFT (contract Owner, counter to 0...)
- *
- * @param args - Args object serialized as a string containing:
- * - the token name (string)
- * - the token symbol (string).
- * - the totalSupply (u64)
- * - the baseURI (string)
- */
-export function _constructor(args: Args): void {
-  // This line is important. It ensures that this function can't be called in the future.
-  // If you remove this check, someone could call your constructor function and reset your smart contract.
-  assert(Context.isDeployingContract());
+import { u256 } from 'as-bignum/assembly';
 
-  const name = args.nextString().expect('name argument is missing or invalid');
-  const symbol = args
-    .nextString()
-    .expect('symbol argument is missing or invalid');
-  const totalSupply = args
-    .nextU64()
-    .expect('totalSupply argument is missing or invalid');
-  const baseURI = args
-    .nextString()
-    .expect('baseURI argument is missing or invalid');
+export const NAME_KEY = stringToBytes('NAME');
+export const SYMBOL_KEY = stringToBytes('SYMBOL');
 
-  Storage.set(nameKey, name);
-  Storage.set(symbolKey, symbol);
-  Storage.set(totalSupplyKey, u64ToBytes(totalSupply));
-  Storage.set(baseURIKey, baseURI);
-  Storage.set(ownerKey, Context.caller().toString());
-  Storage.set(counterKey, u64ToBytes(initCounter));
-}
+export const BALANCE_KEY_PREFIX = 'BALANCE';
+export const OWNER_KEY_PREFIX = 'OWNER';
+export const ALLOWANCE_KEY_PREFIX = 'ALLOWANCE';
+export const OPERATOR_ALLOWANCE_KEY_PREFIX = 'OPERATOR_ALLOWANCE';
```

Comment:
personnal point of view but if we want save storage we should use shorter keys. one or 2 bytes should be enough. even more when its a prefix that will be writen for each users